gemrb/core/Map.cpp

```diff
@@ -916,10 +916,10 @@ bool Map::DoStepForActor(Actor *actor, int speed, ieDword time) {
 		ClearSearchMapFor(actor);
 
 		PathNode * step = actor->GetNextStep();
-		if (step && step->Next) {
-			//we should actually wait for a short time and check then
-			if (GetBlocked(step->Next->x*16+8,step->Next->y*12+6,actor->size)) {
-				actor->NewPath();
+		if (!step) {
+			if (this->GetBlocked(actor->Pos.x, actor->Pos.y, actor->size)) {
+				// TODO: Avoid bumping and bumped actors moving the same way
+				actor->AddWayPoint(this->FindFarthest(actor->Pos, actor->size, actor->size * 4));
```

Comment:
This approach will break if the actor is already on a waypointed path, since they're evaluated in order. Shouldn't work at all. Also looks inefficient — won't it be checking the searchmap all the time for immobile actors?

Reply:
Doesn't break if the actor is moving since it checks for step == null. Agreed that WalkTo() would be better. About efficiency, I'll do some tests and try to come up with a better solution.